canard.h

```diff
@@ -65,12 +65,24 @@ extern "C" {
 #endif
 
 /// Error code definitions; inverse of these values may be returned from API calls.
-#define CANARD_OK                                   0
-// Value 1 is omitted intentionally, since -1 is often used in 3rd party code
-#define CANARD_ERROR_INVALID_ARGUMENT               2
-#define CANARD_ERROR_OUT_OF_MEMORY                  3
-#define CANARD_ERROR_NODE_ID_NOT_SET                4
-#define CANARD_ERROR_INTERNAL                       9
+typedef enum{
+    CANARD_OK                                      = 0,
+// Value 1 is omitted intentionally, since -1 i    =s often used in 3rd party code
+    CANARD_ERROR_INVALID_ARGUMENT                  = 2,
+    CANARD_ERROR_OUT_OF_MEMORY                     = 3,
+    CANARD_ERROR_NODE_ID_NOT_SET                   = 4,
+    CANARD_ERROR_INTERNAL                          = 9,
+// Error codes for Rx handler    =
+    CANARD_ERROR_RX_INCOMPATIBLE_PACKET            = 10,
+    CANARD_ERROR_RX_WRONG_ADDRESS                  = 11,
+    CANARD_ERROR_RX_NOT_WANTED                     = 12,
+    CANARD_ERROR_RX_NO_STATE                       = 13,
+    CANARD_ERROR_RX_MISSED_START                   = 14,
+    CANARD_ERROR_RX_WRONG_TOGGLE                   = 15,
+    CANARD_ERROR_RX_UNEXPECTED_TID                 = 16,
+    CANARD_ERROR_RX_SHORT_FRAME                    = 17,
+    CANARD_ERROR_RX_BAD_CRC                        = 18
+}CanardError;
```

Comment:
This is probably a misuse of enumeration, please revert this back to defines. Enumerations are incompatible with the negated error codes we return from API calls (MISRA C++ 4-5-2) and require casting to an integer when comparing against returned error codes.